### ISSUE TITLE
Gravity defaults to zero in process material

### ISSUE THREAD

**Issue by QbieShay** [godotengine/godot-proposals#2533]
### Describe the project you are working on
Various VFX

### Describe the problem or limitation you are having in your project
In all the effects that I've made, I have used the gravity property of the process material perhaps 10% of the time, of which half of it was going upwards and not with the same magnitude as 9.8.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
I have to change it every time i make a new process material

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Default gravity of the process material to zero

### If this enhancement will not be used often, can it be worked around with a few lines of script?
One click and 2 keys
I could made an addon with some default, pre-configured particle materias to choose form, but that's a lot more work than changing the default to zero :)

### Is there a reason why this should be core and not an add-on in the asset library?
It's the default particle shader.

**Comment by Xrayez** [comment]
I reset gravity to zero as well most of the time but I don't think that would be a good thing to do by itself. I think we should strive to showcase at least basic functionality of what particles can do from scratch for beginners.

This is what basic particles do in 2D when you assign a new particles material:

![image](https://user-images.githubusercontent.com/17108460/113331630-dcf5c000-9328-11eb-862c-f461160191bf.png)

This what happens when you set gravity to zero (mostly nothing):

![image](https://user-images.githubusercontent.com/17108460/113331713-f3038080-9328-11eb-9e89-38c1724a467c.png)

To a beginner it may seem like particles just don't work at all, and they'd have to tinker with more parameters in order to make particles do at least something.

That said, it may be worth doing something different depending on the motivation/use case behind this proposal.

Most particles I make are:
- Not bound to local coordinates.
- Do not have gravity.
- Have initial velocity and maximum spread.

Basically this:

![image](https://user-images.githubusercontent.com/17108460/113332350-bc7a3580-9329-11eb-8691-ce044fb2b944.png)

Which I think may be a better default option for particles material (excluding the suggestion regarding local coordinates perhaps).

**Comment by QbieShay** [comment]
I think that for showcase purposes we can link to tutorials. I believe it's better for it to be functional first for people that use it all the time, rather than giving a good overview as a one-time perk, especially because it could be done in the docs.

**Comment by golddotasksquestions** [comment]
At first I wanted to immediately thumbup this proposal, as I am terribly annoyed having to reset gravity to 0 nearly every time I use particles in 2D.

On second thought I have to agree with Xrayez. If there is one thing that annoys me a million times more than default settings that don't reflect many of my usecases, it's defaults that don't communicate what is going on. 

Like a player in a game, the Godot user needs feedback of their actions. If the Godot user adds something to scene, or preforms an action, there the editor has to provide crystal clear feedback.

Adding particles and then not seeing any particles even though they are playing is like the worst idea for a solution imho. Documentation and tutorials are in no way a replacement for appropriate feedback. Substituting lacking feedback with tutorials is really worst UI design mistake you can make.
If we take another value like initial velocity, we might end up with the same discussion later, but if there is enough community support it might make sense.

**Comment by QbieShay** [comment]
I have been done more VFX with the time passing, and I am more and more convinced that gravity is getting more in the way than helping, being default at -9.8.

Perhaps we could have a "template particle material" choice with a couple of pre-configured ones?

**Comment by Calinou** [comment]
My issue with defaulting the gravity to 0 is that if you don't change anything else (such as linear velocity), then it's very difficult to see any visual feedback.

> Perhaps we could have a "template particle material" choice with a couple of pre-configured ones?

Maybe https://github.com/godotengine/godot-proposals/issues/2998 could be useful here.

**Comment by QbieShay** [comment]
Linear velocity would be equally a problem for me T_T
What i use mostly is scale over life ..

I like a lot the idea of a wizard though!

**Comment by Calinou** [comment]
I've thought a little more about this proposal. Here are my suggestions for the default ParticlesMaterial in both 3D and 2D:

- *(Property: Old → New)*
- Amount: 8 → **10**
- Direction: (1, 0, 0) → **(0, 1, 0)** (pointing upwards)
- Spread: 45 → **15**
- Initial Velocity: 0 → **5** (both Min and Max)
- Scale Curve: None → **The curve below** (high point at roughly 10%, default tangent positions to allow for easing)
  ![Scale Curve inspector](https://user-images.githubusercontent.com/180032/169669150-3c26850c-c38b-4709-a3d8-2b194541c2aa.png)

The same changes should also be applied to CPUParticles for consistency.

With these changes, you get a pretty decent-looking particle system out of the box.

*CPUParticles3D on the left, GPUParticles3D on the right:*

https://user-images.githubusercontent.com/180032/169669121-168c27f2-d62f-42fe-ac6a-f457d8999fc0.mp4

Scale curve is preferred to color curve to fade particles over their lifetime, since it doesn't require you to enable vertex color on the material to work. Also, it performs slightly better as it allows "old" particles to use less GPU fillrate, which is especially important for transparent particles.

**Comment by QbieShay** [comment]
Up for all of those and up too for local coord's proposal!
